src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php

```diff
@@ -62,12 +66,45 @@ public function __construct(RequestContext \$context, LoggerInterface \$logger =
     {
         \$this->context = \$context;
         \$this->logger = \$logger;
+        self::\$declaredRoutes = {$routes};
```

Comment:
You should check if it's null and then assign so you don't overwrite the current array stack every time.